dask/dataframe/io/parquet/core.py

```diff
@@ -378,10 +377,16 @@ def read_parquet_part(fs, func, meta, part, columns, index, kwargs):
     This function is used by `read_parquet`."""
 
     if isinstance(part, list):
-        dfs = [func(fs, rg, columns.copy(), index, **kwargs) for rg in part]
+        dfs = [
+            func(fs, rg, columns.copy(), index, **toolz.merge(kwargs, kw))
+            for (rg, kw) in part
+        ]
         df = concat(dfs, axis=0)
     else:
-        df = func(fs, part, columns, index, **kwargs)
+        # NOTE: `kwargs` are the same for all parts, while `part_kwargs` may
+        #       be different for each part.
+        rg, part_kwargs = part
+        df = func(fs, rg, columns, index, **toolz.merge(kwargs, part_kwargs))
```

Comment:
Part-specific kwargs cannot be merged with the the common kwargs if multiple "pieces" are being accessed for this output partition.